python/pyspark/pandas/namespace.py

```diff
@@ -2814,9 +2821,26 @@ def to_numeric(arg):
     1.0
     """
     if isinstance(arg, Series):
-        return arg._with_new_scol(arg.spark.column.cast("float"))
+        if errors == "coerce":
+            return arg._with_new_scol(arg.spark.column.cast("float"))
+        elif errors == "raise":
+            scol = arg.spark.column
+            scol_casted = scol.cast("float")
+            cond = scol.isNotNull() & scol_casted.isNull()
+            # Filter out if there are data that satisfy the condition.
+            sdf = arg._internal.spark_frame.select(scol).filter(cond)
+            head_sdf = sdf.head(1)
```

Comment:
Can we avoid launching a job here? e.g., `assert_true(col.isNotNull) & assert_true(casted_col.isNotNull)`

Reply:
I think it's not supported the way implementing `assert` with non-boolean objects ??

I think because the Spark Column `col.isNotNull()` itself is non-boolean object, so the `assert` way is not supported?

```python
>>> assert(scol.isNotNull())
Traceback (most recent call last):
...
ValueError: Cannot convert column into bool: please use '&' for 'and', '|' for 'or', '~' for 'not' when building DataFrame boolean expressions.
```